src/inmanta/data/dataview.py

```diff
@@ -486,68 +508,70 @@ def get_extra_url_parameters(self) -> dict[str, str]:
-                If not given, the subquery selects the first resource id, period.
-            """
-            higher_than_condition: str = f"AND r.resource_id > {higher_than}" if higher_than is not None else ""
-            return f"""
-                SELECT
-                    r.resource_id,
-                    r.attributes,
-                    r.resource_type,
-                    r.agent,
-                    r.resource_id_value,
-                    r.model,
-                    r.environment,
-                    (
-                        CASE WHEN (SELECT r.model < latest_version.version FROM latest_version)
-                            THEN 'orphaned' -- use the CTE to check the status
-                            ELSE r.status::text
-                        END
-                    ) as status
-                FROM resource r
-                JOIN configurationmodel cm ON r.model = cm.version AND r.environment = cm.environment
-                WHERE r.environment = $1 AND cm.released = TRUE {higher_than_condition}
-                ORDER BY resource_id, model DESC
-                LIMIT 1
-            """
-
-        query_builder = SimpleQueryBuilder(
+        new_query_builder = SimpleQueryBuilder(
             select_clause="SELECT *",
             prelude=f"""
-                /* the recursive CTE is the second one, but it has to be specified after 'WITH' if any of them are recursive */
-                /* The latest_version CTE finds the maximum released version number in the environment */
-                WITH RECURSIVE latest_version AS (
+               WITH latest_version AS (
                     SELECT MAX(public.configurationmodel.version) as version
                     FROM public.configurationmodel
                     WHERE public.configurationmodel.released=TRUE AND environment=$1
-                ),
-                /*
-                emulate a loose (or skip) index scan (https://wiki.postgresql.org/wiki/Loose_indexscan):
-                1 resource_id at a time, select the latest (released) version it exists in.
-                */
-                cte AS (
-                    /* Initial row for recursion: select relevant version for first resource */
-                    ( {subquery_latest_version_for_single_resource(higher_than=None)} )
-                    UNION ALL
-                    SELECT next_r.*
-                    FROM cte curr_r
-                    CROSS JOIN LATERAL (
-                        /* Recurse: select relevant version for next resource (one higher in the sort order than current) */
-                        {subquery_latest_version_for_single_resource(higher_than="curr_r.resource_id")}
-                    ) next_r
+                ), versioned_resources AS (
+                    SELECT
+                        rps.*,
+                        CASE
+                            -- TODO: first branch may be redundant
```

Comment:
TODO alert. I forgot to test this myself. Perhaps we should get another round of data with just the second branch?

Reply:
If it turns out to be redundant, a `JOIN` may perform similarly, and would be a lot more conventional/readable.

Reply:
Could you make this change?

Reply:
It turns out to make a significant difference after all. I'll just add a comment to replace the TODO.

![screenshot_2024-03-26-100632](https://github.com/inmanta/inmanta-core/assets/7672159/ad878118-0ee6-4c4b-b5ca-7f3ad0613e1c)

Reply:
dd6786128